examples/vision/vision_transformer/megatron_vit_classification_evaluate.py

```diff
@@ -0,0 +1,124 @@
+
+
+@hydra_runner(config_path="conf", config_name="megatron_vit_classification_evaluate")
+def main(cfg) -> None:
+    logging.info("\n\n************** Experiment configuration ***********")
+    logging.info(f'\n{OmegaConf.to_yaml(cfg)}')
+
+    plugins = []
+    strategy = NLPDDPStrategy(
+        no_ddp_communication_hook=True, find_unused_parameters=False,  # we don't use DDP for async grad allreduce
+    )
+    if cfg.get('cluster_type', None) == 'BCP':
+        plugins.append(TorchElasticEnvironment())
+
+    # trainer required for restoring model parallel models
+    trainer = Trainer(plugins=plugins, strategy=strategy, **cfg.trainer)
+
+    save_restore_connector = NLPSaveRestoreConnector()
+    if os.path.isdir(cfg.model.restore_from_path):
+        save_restore_connector.model_extracted_dir = cfg.model.restore_from_path
+
+    model_cfg = MegatronVitClassificationModel.restore_from(
+        restore_path=cfg.model.restore_from_path,
+        trainer=trainer,
+        save_restore_connector=save_restore_connector,
+        return_config=True,
+    )
+
+    assert (
+        cfg.trainer.devices * cfg.trainer.num_nodes
+        == model_cfg.tensor_model_parallel_size * model_cfg.pipeline_model_parallel_size
+    ), "devices * num_nodes should equal tensor_model_parallel_size * pipeline_model_parallel_size"
+
+    # These configs are required to be off during inference.
+    with open_dict(model_cfg):
+        model_cfg.precision = trainer.precision
+        if trainer.precision != "bf16":
+            model_cfg.megatron_amp_O2 = False
+        model_cfg.sequence_parallel = False
+        model_cfg.activations_checkpoint_granularity = None
+        model_cfg.activations_checkpoint_method = None
+
+    model = MegatronVitClassificationModel.restore_from(
+        restore_path=cfg.model.restore_from_path,
+        trainer=trainer,
+        override_config_path=model_cfg,
+        save_restore_connector=save_restore_connector,
+        strict=True,
+    )
+
+    model.eval()
+
+    val_transform = ClassificationTransform(model.cfg, (model.cfg.img_h, model.cfg.img_w), train=False)
+    val_data = ImageFolder(root=cfg.model.data.imagenet_val, transform=val_transform,)
+
+    # initialize apex DDP strategy
+    def dummy():
+        return
+
+    if trainer.strategy.launcher is not None:
+        trainer.strategy.launcher.launch(dummy, trainer=trainer)
+    trainer.strategy.setup_environment()
+
+    test_loader = DataLoader(val_data, batch_size=cfg.model.micro_batch_size, num_workers=cfg.model.data.num_workers,)
+
+    # get autocast_dtype
+    if trainer.precision in ['bf16', 'bf16-mixed']:
+        autocast_dtype = torch.bfloat16
+    elif trainer.precision in [32, '32', '32-true']:
+        autocast_dtype = torch.float
+    elif trainer.precision in [16, '16', '16-mixed']:
+        autocast_dtype = torch.half
+    else:
+        raise ValueError('precision must be in ["32-true", "16-mixed", "bf16-mixed"]')
```

Comment:
I think there's a utility method for this now

Reply:
moved all to use to tool